plugins/modules/vmware_guest.py

```diff
@@ -1258,9 +1258,14 @@ def configure_cpu_and_memory(self, vm_obj, vm_creation=False):
                 self.configspec.firmware = boot_firmware
                 self.change_detected = True
 
-    def sanitize_cdrom_params(self):
+    def sanitize_cdrom_params(self, is_list):
         cdrom_specs = []
-        expected_cdrom_spec = self.params.get('cdrom')
+        if is_list:
+            expected_cdrom_spec = self.params.get('cdrom')
+        else:
+            # just make a one-element list and handle it identically
+            expected_cdrom_spec = [self.params.get('cdrom')]
```

Comment:
This sanitize function was added for configured list of cdrom parameter, since some parameters added, e.g., "controller_type".  So here if  it's not a list, can bypass here.

Reply:
Or new parameters like "controller_type", etc., can be added to the dict. Then in function "configure_cdrom_helper" can also handle this list with one dict.